pkg/cli/start_jemalloc.go

```diff
@@ -19,7 +19,8 @@
 package cli
 
 // #cgo CPPFLAGS: -DJEMALLOC_NO_DEMANGLE
-// #cgo LDFLAGS: -ljemalloc
+// #cgo !darwin LDFLAGS: -ljemalloc
+// #cgo darwin LDFLAGS: -ljemalloc -u_je_zone_register
```

Comment:
Inconsistency doesn't bother me. I would have only made this change to `server/status/runtime_jemalloc.go` because that is the only one we care about. Also, add a comment explaining what is going on.